dev/benchmarks/macrobenchmarks/test/util.dart

```diff
@@ -0,0 +1,220 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:macrobenchmarks/common.dart';
+import 'package:e2e/e2e.dart';
+import 'package:macrobenchmarks/main.dart' as app;
+
+/// The maximum amount of time considered safe to spend for a frame's build
+/// phase. Anything past that is in the danger of missing the frame as 60FPS.
+///
+/// Changing this doesn't re-evaluate existing summary.
+Duration kBuildBudget = const Duration(milliseconds: 16);
+// TODO(CareF): Automatically calculate the refresh budget (#61958)
+
+typedef DriveCallback = Future<void> Function(WidgetController controller);
+
+void macroPerfTestE2E(
+  String testName,
+  String routeName, {
+  Duration pageDelay,
+  Duration duration = const Duration(seconds: 3),
+  Duration timeout = const Duration(seconds: 30),
+  DriveCallback body,
```

Comment:
Nit: as we're moving away from `flutter_driver`, I wonder if `ControlCallback` is a better name than `DriveCallback` as the callback takes a `WidgetController`. CC @dnfield for suggestion too as I imagine that we'll use this in all future e2e performance tests.

Reply:
Done.